### PR TITLE
[SofaGeneral] Pluginize SofaGeneralSimpleFem

### PR DESCRIPTION
**Pull request by fredroy**
#1529

A priori not breaking.

Usual cleaning (namespaces, pragma),
Move non units test to SofaTests
Add dependency to SofaTopologyMapping

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

File: applications/plugins/SofaTest/AffinePatch_test.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ using namespace modeling;
 An affine movement (rotation and translation) is applied to the borders of a mesh. Test if the points inside have the same affine movement.*/
 
 template <typename _DataTypes>
-struct AffinePatch_test : public Elasticity_test<_DataTypes>
+struct AffinePatch_sofa_test : public Elasticity_test<_DataTypes>
```
Comment (guparan) [inline applications/plugins/SofaTest/AffinePatch_test.cpp:52]: Why this?

Comment (fredroy) [reply applications/plugins/SofaTest/AffinePatch_test.cpp:52]: because there is a test in Flexible  with exactly the same name, hence messing up

Comment (epernod) [reply applications/plugins/SofaTest/AffinePatch_test.cpp:52]: too many tests in SOFA!!

**Pull request merged**